matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/profile/DefaultProfileService.kt

```diff
@@ -68,7 +68,7 @@ internal class DefaultProfileService @Inject constructor(private val taskExecuto
     }
 
     override suspend fun updateAvatar(userId: String, newAvatarUri: Uri, fileName: String) {
-        withContext(coroutineDispatchers.main) {
+        withContext(coroutineDispatchers.io) {
```

Comment:
this is the fix

Reply:
Its something we already had fixed, not sure why its back. But fileUploader should already dispatch on a background thread, its weird

Reply:
yeah it's very strange, I checked the okhttp thread pool/interceptors as well, couldn't find any cases where the main thread was being used, unless on some devices the okhttp thread uses the main thread/looper for some reason :thinking:

Reply:
This is also something which has been fixed by some fork on their side FWIW.